akka-persistence/src/main/scala/akka/persistence/fsm/PersistentFSM.scala

```diff
@@ -111,21 +124,32 @@ trait PersistentFSM[S <: FSMState, D, E] extends PersistentActor with Persistent
       //Persist the events and apply the new state after all event handlers were executed
       var nextData: D = stateData
       var handlersExecutedCounter = 0
+      var doSnapshot: Boolean = false
+      val moduloCondition: Long ⇒ Boolean = snapshotAfter match {
+        case Some(snapShotAfterValue) ⇒ lastSequenceNr: Long ⇒ lastSequenceNr % snapShotAfterValue == 0
```

Comment:
Followed the advice in the new commit, although the code was moved to `SnapshotAfter`, a class of Akka Extension.